packages/components/src/filters/advanced/number-filter.js

```diff
@@ -0,0 +1,184 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { Component, Fragment } from '@wordpress/element';
+import { SelectControl, TextControl } from '@wordpress/components';
+import { get, find, partial } from 'lodash';
+import interpolateComponents from 'interpolate-components';
+import classnames from 'classnames';
+import { _x } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import TextControlWithAffixes from '../../text-control-with-affixes';
+import { formatCurrency } from '../../../../currency/src';
+
+class NumberFilter extends Component {
+	getBetweenString() {
+		return _x(
+			'{{moreThan /}} and {{lessThan /}}',
+			'Numerical range inputs arranged on a single line',
+			'wc-admin'
+		);
+	}
+
+	getLegend( filter, config ) {
+		const inputType = get( config, [ 'input', 'type' ], 'number' );
+		const rule = find( config.rules, { value: filter.rule } ) || {};
+		let filterStr = filter.value || '';
+
+		if ( 'between' === rule.value ) {
+			let [ moreThan, lessThan ] = filterStr.split( ',' );
+
+			if ( 'currency' === inputType ) {
+				moreThan = formatCurrency( moreThan );
+				lessThan = formatCurrency( lessThan );
+			}
+
+			filterStr = interpolateComponents( {
+				mixedString: this.getBetweenString(),
+				components: {
+					moreThan: <Fragment>{ moreThan }</Fragment>,
+					lessThan: <Fragment>{ lessThan }</Fragment>,
+				},
+			} );
+		} else if ( 'currency' === inputType ) { // need to format a single input value as currency
+			filterStr = formatCurrency( filterStr );
+		}
+
+		return interpolateComponents( {
+			mixedString: config.labels.title,
+			components: {
+				filter: <span>{ filterStr }</span>,
+				rule: <span>{ rule.label }</span>,
+			},
+		} );
+	}
+
+	getFormControl( type, value, onChange ) {
+		if ( 'currency' === type ) {
+			const currencySymbol = get( wcSettings, [ 'currency', 'symbol' ] );
+			const symbolPosition = get( wcSettings, [ 'currency', 'position' ] );
+
+			return (
+				0 === symbolPosition.indexOf( 'right' )
+				? <TextControlWithAffixes
```

Comment:
Do you think it's worth adding an `.is-error` class if moreThan > lessThan?

Reply:
🤔 

As long as they aren't equal, we can form a range out of the values - so long as "_ is between 10 and 1" isn't confusing. Allowing that would still need an error case for the values being equal. If we do allow the greater value in the first input, that could simplify the `.split()` logic when changing filter rules (just always take the first value when changing from "between").

I looked and didn't see any project-wide `.is-error` styling, just some one-off uses in the calendar, pagination, and summary. Can we add generic component error styling or does it all need to be done on a per-component basis?

Reply:
Discussed this with @timmyc and we decided to allow non-positional range values for the "between" rule. (E.g. "between 200 and 100")

Punting on the error handling for now (which would only be checking that range start != range end).